jerry-core/jerry.c

```diff
@@ -1698,6 +1698,8 @@ jerry_parse (const jerry_api_char_t *source_p, /**< script source */
 {
   jerry_assert_api_available ();
 
+  vm_finalize ();
```

Comment:
I didn't touch `jerry_parse_and_save_snapshot`. I didn't want to introduce too big changes to the embedder (main-unix.c) -- it might have required command line interface change to enable saving multiple snapshots of multiple input script files. That's why if snapshot saving is enabled, only a single input JS script is allowed, see below:
```c
    if (files_counter != 1)
    {
      jerry_port_errormsg ("Error: --save-snapshot argument works with exactly one script\n");
      return JERRY_STANDALONE_EXIT_CODE_FAIL;
    }
```
Thus, as invokation count of `jerry_parse_and_save_snapshot` didn't increase, I didn't examine/touch that function either.